pkg/agent/client.go

```diff
@@ -538,10 +538,19 @@ func (a *Client) remoteToProxy(connID int64, eConn *endpointConn) {
 	}
 
 	for {
+		select {
+		case <-a.stopCh:
+			return
+		default:
+		}
+		timeout := time.Now().Add(15 * time.Second)
+		eConn.conn.SetReadDeadline(timeout)
 		n, err := eConn.conn.Read(buf[:])
 		klog.V(5).InfoS("received data from remote", "bytes", n, "connectionID", connID)
 
-		if err == io.EOF {
+		if err == os.ErrDeadlineExceeded {
+			continue
```

Comment:
Can you add a unit test for this case? Maybe put the timeout in a global variable that can be overridden for a faster test. In particular, I want to make sure that a streaming connection that doesn't send any data for a window longer than the timeout won't be broken.

Reply:
Would you be good using the delayedServer in the proxy_test for this? Currently it waits between 500ms and 2 seconds. I could bump that up to waiting between 5 and 10 seconds. Then as you suggest shortening the timeout for the read on the test from 15 seconds down to 1 second.

Reply:
Done